src/web/components/Core/PagingTool.tsx

```diff
@@ -42,14 +44,14 @@ export function PagingTool({
   }, [numberTotalRows, initialRowsPerPage, initialPageNumber]);
 
   const onIncreasePageNumber = () => {
-    if (pageNumber < Math.ceil(numberTotalRows / rowsPerPage)) {
+    if (canIncreasePageNumber(pageNumber, numberTotalRows, rowsPerPage)) {
```

Comment:
I think we can actually move this condition to the disabled state of the button - then we won't need to check it here in the `onClick`? i.e. 
```
const onIncreasePageNumber = () => {
  setPageNumber(pageNumber + 1);
  onChangeRows(pageNumber + 1, rowsPerPage);
};
...
<button
  type='button'
  className='icon-button'
  title='Next Page'
  onClick={onIncreasePageNumber}
  disabled={!canIncreasePageNumber(pageNumber, numberTotalRows, rowsPerPage)}
>
  <FontAwesomeIcon icon='angle-right' />
</button>
```

How's that sound?

Reply:
then ditto for decrementing the page number